web/assets/js/components/gameboard.ts

```diff
@@ -1,232 +1,247 @@
-
-    const currentCell = e.currentTarget.dataset.square;
-
-    if (this.moves.includes(currentCell)) {
-      this.moves = [];
-      let promoted: string | number = 0;
-
-      //if promotion, wwait for user choice
-      if (this.promotion.length > 0) {
-        this.DOM.board.style.pointerEvents = "none";
-        promoted = await this.promote();
-        this.DOM.board.style.pointerEvents = "auto";
-      }
-
-      // if click on allowed cell (to move)
-      const move = this.chess.move({ from: this.selected, to: currentCell, promotion: promoted });
-      this.board.move(`${move.from}-${move.to}`);
-      if (move.captured) {
-        // capture
-        this.call("capturePawn", [move.captured], "gameplayers", move.color === this.color ? "me" : "rival");
-      }
-
-      // ------ Action - emmit my move ------
-      await Action.makeMove(move.from, move.to, promoted);
-      this.engine();
-
-      //reset allowed positions
-      gsap.to(".chess-board [data-square]", { "--disp-opacity": 0 });
-      this.DOM.moves = [];
-    } else {
-      // if click on disallow cell (new select)
-      if (this.DOM.moves.length > 0) {
-        // if new select on board -> reset allowed position
-
-        gsap.to(".chess-board [data-square]", { "--disp-opacity": 0 });
-        this.DOM.moves = [];
-      }
-
-      // get allowed position on board DOM then DOM
-      this.selected = currentCell;
-      const moves = this.chess.moves({ square: currentCell, verbose: true });
-      this.moves = [...new Set(moves.map((pos: any) => pos.to))]; //remove duplicate of "to" (promotion ones)
-      this.promotion = [...new Set(moves.filter((move: any) => move.promotion).map((move: any) => move.to))];
-
-      this.moves.forEach((cell: Element) => {
-        this.DOM.moves.push(this.DOM.el.querySelector(`[data-square="${cell}"]`));
-      });
-      // highlight allowed position
-      if (this.DOM.moves.length > 0) gsap.to(this.DOM.moves, { "--disp-opacity": 1 });
+
+    showScoreBoard(winner: Colors) {
+        gsap.set(this.DOM.el, {willChange: "transform"});
+        gsap.to(this.DOM.el, {
+            rotate: winner === "w" ? "-20deg" : "20deg",
+            x: "60%",
+            y: winner === "w" ? "0%" : "25%",
+            scale: 0.8
+        });
+    }
+
+    getMoveNumber() {
+        return this.chess.moveNumber();
+    }
+
+    async engine(init = false, gameover?: GameoverType) {
+        //for test purpose -> to et in if statment bellow
+        // setTimeout(() => {
+        //   this.call("finishGame", "gameover", "gameplayers", "me");
+        //   this.call("stopTimer", [true], "gameplayers", "me");
+        //   this.call("stopTimer", [true], "gameplayers", "rival");
+        //   this.call("disappear", "", "gamecontrols");
+        //   this.showScoreBoard("w");
+        // }, 10000);
+
+        const actions: Actions = await Actions.getInstance()
+
+        //TODO: win cause rival resign
+        //TODO: draw screen in players
```

Comment:
We should add a check game status system here to sync the game backend with the front. cause there is other `gameover` status only the backend know (cf: https://allinbits.slack.com/archives/C05NU14R0R2/p1695041304135699?thread_ts=1695037716.680729&cid=C05NU14R0R2)

Reply:
You can fetch the game easily using Actions.getGame(gameID), and check the game.state against end states. This call is cheap and fast (doesn't require a transaction commit), so you can do it on each move since event emissions are still a revolutionary concept